### ISSUE TITLE
Inconsistent name detection for theme variables

### ISSUE THREAD

**Issue by sclu1034** [awesomeWM/awesome#3718]
@Elv13 

The "Default value" and "fallback when `nil`" fields are inconsistent in how they determine the `beautiful` variable name.

![image](https://user-images.githubusercontent.com/4508454/195055677-6d08364a-2c55-4b4a-929d-4784bb7ddc85.png)

> https://awesomewm.org/apidoc/widgets/wibox.widget.separator.html

Besides, the "default value" is incorrect anyways. While the draw call will fall back to the theme variable, the getter does not, so the property itself will return `nil`.

**Comment by Elv13** [comment]
Yes. There is room for improvements there. The getter part should probably be treated as a minor code bug rather than a doc bug.

As for the rendering consistency itself, I am undecided on how to tackle it properly. I was aware of this issue when I wrote it, but I consider having this documented (which is correct the majority of the time) better than the ambiguity it replaces. One option I am considering is to collapse the `beautiful` documentation just like we collapsed signals doc into the property. The theme variable doc is mostly useless since it has no or non-useful description with no examples or context. Just like signals, properties fallback variables could be documented inline. That should come after the "appearance" doc page is rewritten, right now it is implemented as a regex on top of the code rather than as part of the documentation generation "proper", which means it is very fragile and only works when the doc is written exactly as-is. This preclude doing the changes until that's fixed.
